pages/_app/index.jsx

```diff
@@ -110,12 +110,21 @@ class App extends NextApp {
         })
 
         // Subscribe store to router after initialization
-        router.events.on('routeChangeStart', this.handleRouteChange)
+        // We use routeChangeComplete otherwise router.query contains previous
+        // values
+        router.events.on('routeChangeComplete', this.handleRouteChange)
+
+        try {
+          store.changeDataProvider(dataProviderId)
+        } catch (error) {
+          if (error instanceof AccessError)
+            await router.replace('/data-providers')
+          else throw error
+        }
```

Comment:
If I understand this correctly, when `then`'s callback throws, the error is caught but the following `catch`. You can handle all the errors in a single place.

Reply:
Check it now.